src/crawlers/playwright_crawler.js

```diff
@@ -0,0 +1,282 @@
+ *  Only available if used by the crawler.
+ * @property {?RequestQueue} requestQueue
+ *  A reference to the underlying {@link RequestQueue} class that manages the crawler's {@link Request}s.
+ *  Only available if used by the crawler.
+ * @property {?SessionPool} sessionPool
+ *  A reference to the underlying {@link SessionPool} class that manages the crawler's {@link Session}s.
+ *  Only available if used by the crawler.
+ * @property {?ProxyConfiguration} proxyConfiguration
+ *  A reference to the underlying {@link ProxyConfiguration} class that manages the crawler's proxies.
+ *  Only available if used by the crawler.
+ * @property {BrowserPool} browserPool
+ *  A reference to the underlying `BrowserPool` class that manages the crawler's browsers.
+ *  For more information about it, see the [`browser-pool` module](https://github.com/apify/browser-pool).
+ * @property {AutoscaledPool} autoscaledPool
+ *  A reference to the underlying {@link AutoscaledPool} class that manages the concurrency of the crawler.
+ *  Note that this property is only initialized after calling the {@link CheerioCrawler#run} function.
+ *  You can use it to change the concurrency settings on the fly,
+ *  to pause the crawler by calling {@link AutoscaledPool#pause}
+ *  or to abort it by calling {@link AutoscaledPool#abort}.
+ */
+class PlaywrightCrawler extends BrowserCrawler {
+    static optionsShape = {
+        ...BrowserCrawler.optionsShape,
+        browserPoolOptions: ow.optional.object,
+        playwrightModule: ow.optional.object,
+        gotoTimeoutSecs: ow.optional.number,
+        launchOptions: ow.optional.object,
+        gotoFunction: ow.undefined,
+    }
+
+    constructor(options = {}) {
+        ow(options, 'PlaywrightCrawlerOptions', ow.object.exactShape(PlaywrightCrawler.optionsShape));
+
+        const {
+            playwrightModule,
+            launchContext = {},
+            gotoTimeoutSecs,
+            browserPoolOptions = {},
+            ...browserCrawlerOptions
+        } = options;
+
+        browserPoolOptions.browserPlugins = [
+            new PlaywrightPlugin(
+                // eslint-disable-next-line
+                getPlaywrightLauncherOrThrow(playwrightModule),
+                {
+                    launchOptions: apifyOptionsToLaunchOptions(launchContext),
+                },
+            ),
+        ];
+
+        super({
+            ...browserCrawlerOptions,
+            browserPoolOptions,
+        });
+
+        this.browserPool.postLaunchHooks.push(({ error, session }) => {
+            // It would be better to compare the instances,
+            if (error && error.constructor.name === 'TimeoutError') {
+                handleRequestTimeout(session, error.message);
+            }
+        });
+
+        this.gotoTimeoutMillis = gotoTimeoutSecs * 1000;
+
+        this.launchContext = launchContext;
+        this.playwrightModule = playwrightModule;
```

Comment:
launcher. Maybe we could even put the launcher to `BrowserCrawlerOptions`

Reply:
I think it is a crawler property. Since the crawler configures the plugin.